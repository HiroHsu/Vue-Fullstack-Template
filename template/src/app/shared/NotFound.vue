<script>
export default {
  metaInfo: {
    title: 'Page Not Found | Vue by Example',
  },
  render() {
    return (
      <v-layout class="vfs-notfound">
        <h1 class="pa-3">404</h1>
      </v-layout>
    );
  },
};
</script>

<style scoped>
.vfs-notfound {
  position: absolute;
  left: 0;
  top: 0;
  z-index: 999;
  width: 100%;
  height: 100%;
  background: #fff;
}
</style>
