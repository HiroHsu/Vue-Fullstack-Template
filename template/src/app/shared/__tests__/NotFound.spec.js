import Vuex from 'vuex';
import { createLocalVue, shallow } from '@vue/test-utils';

import NotFound from '../NotFound';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('Not Found', () => {
  let [wrapper, store] = [];

  beforeEach(() => {
    store = new Vuex.Store({
      state: {
        theme: 'light',
      },
    });

    wrapper = shallow(NotFound, { localVue, store });
  });

  it('should render light component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render dark component', () => {
    store.state.theme = 'dark';
    expect(wrapper.html()).toMatchSnapshot();
  });
});
