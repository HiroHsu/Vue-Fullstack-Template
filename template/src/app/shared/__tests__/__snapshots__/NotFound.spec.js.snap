// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound should render initial component 1`] = `
<v-layout class="vfs-notfound">
  <h1 class="pa-3">404</h1>
</v-layout>
`;
